src/kraft_ws_util.erl

```diff
@@ -78,18 +78,22 @@ websocket_init(#{conn := Conn} = State0) ->
     module(init, [Conn], State1).
 
 websocket_handle(pong, #{ping := #{}} = State0) ->
-    {[], State0};
+    State1 = cancel_pong_timeout(State0),
+    {[], State1};
 websocket_handle(Frame, State0) ->
     module(handle, [Frame], State0).
 
 websocket_info('$kraft_ws_ping', State0) ->
-    State1 = trigger_ping(State0),
+    State1 = trigger_pong_timeout(trigger_ping(State0)),
     {[ping], State1};
+websocket_info('$kraft_ws_pong_timeout', State0) ->
+    {[{close, 1011, <<"timeout waiting for pong">>}], State0};
 websocket_info(Info, State0) ->
     module(info, [Info], State0).
 
 terminate(Reason, Req, State0) ->
     cancel_ping(State0),
+    cancel_pong_timeout(State0),
     module(terminate, [Reason, Req], State0),
     ok.
 
```

Comment:
The default opts need to be updated, for example...
`default_opts() -> #{type => raw, ping => #{interval => 30_000}, pong => #{timeout => 1_000}}.`

Reply:
@maehjam you missed this during the rebase

Reply:
Then I also get this 
```
=WARNING REPORT==== 28-May-2024::15:35:01.362930 ===
Websocket unhandled frame: pong
```
I'll do a PR for this branch with the fix